opslevel/terraform_type_conversions.go

```diff
@@ -57,3 +58,15 @@ func ListValueToStringSlice(ctx context.Context, listValue basetypes.ListValue)
 	diags := listValue.ElementsAs(ctx, &dataAsSlice, true)
 	return dataAsSlice, diags
 }
+
+// Converts a basetypes.MapValue to an opslevel.JSON
+func MapValueToOpslevelJson(ctx context.Context, mapValue basetypes.MapValue) (opslevel.JSON, diag.Diagnostics) {
+	mapAsJson := opslevel.JSON{}
+	stringMap := map[string]string{}
+
+	diags := mapValue.ElementsAs(ctx, &stringMap, false)
+	for k, v := range stringMap {
+		mapAsJson[k] = v
+	}
+	return mapAsJson, diags
```

Comment:
Could you also just do `return opslevel.JSON(mapAsJson), diags` here?

Reply:
As in, change `mapAsJson := opslevel.JSON{}` to `mapAsJson := map[string]any{}` then type cast mapAsJson on the way out? That would work, but I can also skip the type casting by setting the correct type from the start

Reply:
@davidbloss actually it probably won't work since `string != any` -- doesn't matter since it's the same result anyways